app/controllers/posts_controller.rb

```diff
@@ -4,7 +4,15 @@ class PostsController < ApplicationController
   before_action :correct_user, only: %i(edit update)
 
   def index
-    @posts = Post.order(created_at: :desc).page(params[:page]).per Settings.posts.page.max
+    @topics = Topic.all
+
+    top = params[:top]
+
+    if !top.nil?
+      @posts = Post.where(:topic_id => top).page(params[:page]).per Settings.posts.page.max
+    else
+      @posts = Post.all.order(created_at: :desc).page(params[:page]).per Settings.posts.page.max
+    end
```

Comment:
I think you should change code from 11 line - 15 line to
- in Post model file
add code:
`scope :lastest, ->{order created_at: :desc}`
`scope :by_top, ->(top_id){where top_id: top_id if top_id.present?}`
- in here, change code from 11 line to 15 line:
@posts = Post.by_top(top).lastest.page(params[:page]).per Settings.posts.page.max